### PR TITLE
Fixed comments round 1

### PR DESCRIPTION
**Pull request by kkovrizhenko**
1. Calculator needs to run continuously till the user types 'exit' command (not after single computation)

2. If we add two integers (e.g. 1+1) we expect result '2', not '2.0'

3. A good practice is to avoid naming methods and variables similar to some builtins - e.g. python 'input' and your method 'inputs' (you can use some more complex names like 'read_user_input')

4. 'Return' in line 42 is redundant - will never be executed

5. start() in line 84 can be wrapped by: "if __name__ == '__main__':" - this is not required but it's also a kind of convention, especially important when you will import this file in some other. Probably we will have this problem when writing tests (https://stackoverflow.com/questions/419163/what-does-if-name-main-do)

6. "def start() -> object:" => actually we use typing in brainy for writing tests. Part "-> object" says that function "start" returns an "object". In your calculator "start" doesn't return anything so much better is to write "-> None". Lets add also return types to other functions

7. Lets add also docstrings for each function. Some example:
def do_the_math(factor_x: str, factor_y: str, math_operator: str) -> float:
    """
    Function that does the calculation and print the result
    :param factor_x: first number for math calculation
    :param factor_y: second number for math calculation
    :param math_operator: mathematical operator: (+, -, *, /)
    """

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -26,59 +31,79 @@ def validate_number(num):
...
-            inputs()
+            read_user_inputs()
     elif operator == '*':
-        print("The result of multiplying: ", round(first_num * second_num, 3))
+        print("The result of multiplying: ", do_math(round(first_num * second_num, 3)))
+
 
+def do_math(num: T) -> T:
+    """Check if the number is integer and convert it"""
+    return int(num) if num%1 == 0 else num
```
Comment (marcinoleksiak) [inline main.py]: This method actually doesn't do anything right now? no matter what number you take, this number % 1 will be this number (dividing by one)

File: main.py [hunk 2]
```diff
@@ -1,14 +1,18 @@
 # This is a “Console calculator" for a Python AQA Bootcamp task
 import sys
+from typing import TypeVar, Callable
 
 valid_operators = ['+', '-', '/', '*']
+T = TypeVar('T', int, float)
```
Comment (marcinoleksiak) [inline main.py]: This is a nice trick, which I like the most in your version of calculator :)
But you don't have to use such advanced structures in this simple implementation. 

(this is sth definitely worth remembering)

File: main.py [hunk 3]
```diff
@@ -17,7 +21,8 @@ def instruction():
           "\n - Number can be a float number (6.66)")
 
 
-def validate_number(num):
+def validate_number(num: any) -> bool:
```
Comment (marcinoleksiak) [inline main.py]: you don't have to use 'any' here. you're validating if passed string is a number.
so:
(num: str)
is enough here :)

File: main.py [hunk 4]
```diff
@@ -26,59 +31,79 @@ def validate_number(num):
         return False
 
 
-def validate_operator(operator):
+def validate_operator(operator: any) -> bool:
```
Comment (marcinoleksiak) [inline main.py]: similar thing is here. you're getting a string in argument so: (operator: str) is enough
if we decided to add all this typing let's make it as much accurate as possible

File: main.py [hunk 5]
```diff
@@ -26,59 +31,79 @@ def validate_number(num):
...
+    """Validate value for being in a list of operators and print a massage if the validation failed."""
     if operator in valid_operators:
         return True
 
     print(f'Please enter one on these operators: {", ".join(valid_operators)}')
     return False
 
 
-def handle_input(phrase, validator):
+def handle_input(phrase: str, validator: Callable[[any], bool]) -> T:
```
Comment (marcinoleksiak) [inline main.py]: according to previous comments a small modification here: validator: Callable[[str], bool]

File: main.py [hunk 6]
```diff
@@ -26,59 +31,79 @@ def validate_number(num):
...
+def read_user_inputs() -> None:
+    """Ask user to enter 2 numbers and operators, process them and show result, rinse and repeat"""
     a = handle_input("Enter first number:", validate_number)
     operator = handle_input("Enter the operator you want to use:", validate_operator)
     b = handle_input("Enter second number:", validate_number)
     count(a, b, operator)
+    read_user_inputs()
+
 
+def count(a: T, b: T, operator: str) -> None:
```
Comment (marcinoleksiak) [inline main.py]: this is a place when we can remove this type T. Note that when you're calling count() method, you're passing 3 arguments: a,b,operator that are all strings. No need to say that 'a' can be int or float. The same for b.
replace 'T' with 'str' here also

File: main.py [hunk 7]
```diff
@@ -26,59 +31,79 @@ def validate_number(num):
...
+    """
+    Perform math operation based on user inputs and print result
 
-def count(a, b, operator):
+    :param a: first number
+    :param b: second number
+    :param operator: math operator
+    """
     first_num = float(a)
     second_num = float(b)
```
Comment (marcinoleksiak) [inline main.py]: This is sth that confirms my previous comment. Despite you've declared that count() accepts argument 'a' that can be [int, float], you need to cast it again to float.

In this moment I'd like to suggest a bit other approach for calculating the result. Check this out:
image

the eval() function does the job for you. also checking for division by 0 is being done automatically by catching the exception. The only thing that left here is to change the way you present the outcome.
What do you think?

File: .gitignore [hunk 8]
```diff
@@ -1 +1,2 @@
-/venv
\ No newline at end of file
+/venv
+.idea
```
Comment (marcinoleksiak) [inline .gitignore]: you can find some default .gitignore file contents for most of languages, if not for all of them :)
the one for python: https://github.com/github/gitignore/blob/master/Python.gitignore

**Review by marcinoleksiak — CHANGES_REQUESTED** [review]
Hey, I think we are almost in the end of this part. Please, check my comment and let me know what you think :)

File: main.py (hunk 1 above)
Comment (kkovrizhenko) [reply main.py]: @marcinoleksiak actually, I tried to use here modulo to check if the result of the math operation is int or float, to not display .00 in the final result if they are not needed
got the inspiration here - https://stackoverflow.com/questions/4541155/check-if-a-number-is-int-or-float/44349980#44349980
or is it not best solution?

Comment (marcinoleksiak) [reply main.py]: actually it's a good one. I didn't notice the trick :)

**Review by marcinoleksiak — APPROVED** [review]

**Pull request merged**